core/src/main/java/org/elasticsearch/common/settings/IndexScopedSettings.java

```diff
@@ -133,8 +134,15 @@
         FsDirectoryService.INDEX_LOCK_FACTOR_SETTING,
         EngineConfig.INDEX_CODEC_SETTING,
         IndexWarmer.INDEX_NORMS_LOADING_SETTING,
-        // this sucks but we can't really validate all the analyzers/similarity in here
-        Setting.groupSetting("index.similarity.", false, Setting.Scope.INDEX), // this allows similarity settings to be passed
+        // validate that built-in similarities don't get redefined
+        Setting.groupSetting("index.similarity.", false, Setting.Scope.INDEX, (s) -> {
+          boolean valid = true;
+          String similarityName = s.substring(0, s.indexOf("."));
+          if(SimilarityService.BUILT_IN.keySet().contains(similarityName)) {
+            throw new IllegalArgumentException("Cannot redefine built-in Similarity [" + similarityName + "]");
+          }
+          return valid;
+        }), // this allows similarity settings to be passed
```

Comment:
@s1monw here is an attempt at validating similarities as part of  `IndexScopedSettings.BUILT_IN_INDEX_SETTINGS`. Non-group settings seem to leverage their parsers to perform validation but group settings don't have that ability. I added the ability to pass a `Predicate` to `Settings.groupSetting()` so that some validation can be performed as part of `Setting.get()`. Predicates can be defined in `IndexScopedSettings` or somewhere else but the main issue I have with this approach is finding a static way to verify the current index version. Let me know what you think.